ntbs-service-unit-tests/DataAccess/DataQualityRepositoryTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using ntbs_service.DataAccess;
+using ntbs_service.Models;
+using ntbs_service.Models.Entities;
+using ntbs_service.Models.Entities.Alerts;
+using ntbs_service.Models.Enums;
+using Xunit;
+
+namespace ntbs_service_unit_tests.DataAccess
+{
+    // This suite attempts to test EF queries, close to the context. It draws on this article for the setup inspiration
+    // (albeit without the extensive multi-provider support):
+    // https://docs.microsoft.com/en-us/ef/core/miscellaneous/testing/testing-sample
+    public class DataQualityRepositoryTests
+    {
+        private const string NoAlertsName = "A";
+        private const string BirthCountryAlertName = "B";
+        private const string ClinicalDatesAlertName = "C";
+
+        public DataQualityRepositoryTests()
+        {
+            ContextOptions = new DbContextOptionsBuilder<NtbsContext>()
+                .UseInMemoryDatabase(nameof(DataQualityRepositoryTests))
+                .Options;
+        }
+
+        private DbContextOptions<NtbsContext> ContextOptions { get; }
+
+        [Fact]
+        public async Task FiltersOutNotificationsWithTheAlertTypeAlreadyPresent()
+        {
+            // ARRANGE
+            using (var context = new NtbsContext(ContextOptions))
+            {
+                await context.Database.EnsureDeletedAsync();
+                await context.Database.EnsureCreatedAsync();
```

Comment:
Hmm, I guess they're not technically. They were in the code I used as the basis for this, where this setup was happening in a `beforeEach` block, so it was responsible fore resetting the db between each test. It also accommodated other db types (including persistent ones).

I kind of like leaving them there so that puzzle doesn't need solving by the person who adds a second test to this suite - what do you think?